native/chat/chat-thread-list.react.js

```diff
@@ -43,13 +43,17 @@ const floatingActions = [
   },
 ];
 
-type Item = ChatThreadItem | {| type: 'search', searchText: string |};
+type Item =
+  | ChatThreadItem
+  | {| type: 'search', searchText: string |}
+  | {| type: 'empty', emptyItem: React.Node |};
 
 type Props = {|
   navigation: ChatTopTabsNavigationProp<
     'HomeChatThreadList' | 'BackgroundChatThreadList',
   >,
   filterThreads: (threadItem: ThreadInfo) => boolean,
+  emptyItem: React.Node,
```

Comment:
Since this is an optional property, let's specify it as:

```js
  emptyItem?: React.Node,
```

(Flow doesn't catch this because the type of our `connect` function is broken, and I have never got around to fixing it...)